### DISCUSSION TITLE
Precognition vary header

### DISCUSSION THREAD

**Discussion by apih** [laravel/framework#52111] [Q&A] [answered]
Hi, just want to ask, why does the [`HandlePrecognitiveRequests` middleware](https://github.com/laravel/framework/blob/11.x/src/Illuminate/Foundation/Http/Middleware/HandlePrecognitiveRequests.php) append the `Vary` header on the response, with _Precognition_ value, as seen on [this line](https://github.com/laravel/framework/blob/f5f9383af919ef74b00eacb23b4a88fcf390743c/src/Illuminate/Foundation/Http/Middleware/HandlePrecognitiveRequests.php#L41), even when the request is not attempting precognition?

Maybe @timacdonald can share the reason?

**Comment by timacdonald** [comment]
Hey, @apih, the Vary header is used as a cache key to indicate that a page is different.

It needs to be on all requests for the given URL.

You can read more about it in the MDN docs: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Vary

Hope that helps.

**Reply by apih** [reply]
I am using `HandlePrecognitiveRequests` middleware on a resource route right now, like below, and I can see the Vary header on GET requests too.

```php
Route::resource('products', ProductController::class)->middleware(HandlePrecognitiveRequests::class);
```

There are only examples for normal routes, explicitly for POST (data submission), in [Precogntion documentation](https://laravel.com/docs/11.x/precognition), so I am not sure if using the middleware on resource routes is right or not.

**Reply by timacdonald** [reply, marked as answer]
You may use Precognition on any request method.

**Reply by apih** [reply]
I see, thanks for the clarification.

**Reply by timacdonald** [reply]
No troubles at all.
